### DISCUSSION TITLE
Is there an option to disable the image magick plugin?

### DISCUSSION THREAD

**Discussion by GalaxySnail** [sxyazi/yazi#1530] [1. Q&A] [answered]
I have an old laptop with a weak Intel N450 CPU. When I enter a direcotry contains some images, yazi executes `magick` which costs 100% CPU and a lot of memory for a long time. My workaround is:

```shell
tmp=$(mktemp -d)
ln -s /usr/bin/false $tmp/magick
PATH="$tmp:$PATH" yazi
```

And yazi works well. Is there any way to completely disable the `magick` plugin without the `$PATH` hack?

**Comment by lpnh** [comment]
See if the following solution works for you: https://github.com/sxyazi/yazi/issues/735#issuecomment-1965659093

**Reply by GalaxySnail** [reply]
This completely disables image preview, but I do want to preview images. By masking `magick`, I can still preview images after cache cleared, it seems that `magick` isn't required.

**Reply by sxyazi** [reply]
Just change the mime to match that `magick` one:

https://github.com/sxyazi/yazi/blob/4fa37cb3cd001f1bfadffba553b071f56fbc8624/yazi-config/preset/yazi.toml#L96

```toml
[[plugin.prepend_preloaders]]
mime = "image/{avif,hei?,jxl,svg+xml}"
run  = "noop"
```

**Reply by GalaxySnail** [reply, marked as answer]
Unfortunately it doesn't work, yazi is still calling `magick`. It seems that both `prepend_preloaders` and `prepend_previewers` need to be set:

```toml
[plugin]
prepend_preloaders = [
    { mime = "image/{avif,hei?,jxl,svg+xml}", run = "noop" }, 
]
prepend_previewers = [
    { mime = "image/{avif,hei?,jxl,svg+xml}", run = "noop" }, 
]
```

and it works, yazi no longer calls `magick`. If I understand correctly, everytime the mime type in yazi's default configuration changes, I need to change my config to adapt to it. So I choose to use this config and it also works:

```toml
[plugin]
prepend_preloaders = [
    { mime = "image/*", run = "image" },
]
prepend_previewers = [
    { mime = "image/*", run = "image" },
]
```

Thanks for your help!

**Comment by TobisLee** [comment]
Same issue here, and it crashes `vscode`. I prefer to limit CPU and memory usage. I tried adjusting the `image_alloc` size and setting `multi=false` in the preloader, but `vscode` still crashes. I don't mind if loading image previews takes more time—I just want to keep the image preview size without crashing apps. Is there any solutions?
